gitflow.qmd

```diff
@@ -0,0 +1,65 @@
+# Gitflow {{< iconify fa6-solid code-branch >}} {#sec-gitflow}
+
+[Git][git] is the version control system that underlies GitHub. Gitflow is an alternative git branching model that uses feature branches and multiple primary branches (vs one primary branch, e.g., "main").
+
+Gitflow was originally proposed by [Vincent Driessen in 2010 on his blog][gitflowblog]. [A tutorial][gitflowtut] by the tech company Atlassian has a nice overview of Gitflow.
+
+In Gitflow there are two primary branches: `main` and `dev` (see diagram below). Feature branches always branch off of `dev`. After feature branches merge back into `dev`, then `dev` is merged into `main`.
+
+```{mermaid}
+%%{init: { 'logLevel': 'debug', 'theme': 'default', 'themeVariables': {
+  'git0': '#57bfd3',
+  'git1': '#5F9157',
+  'git2': '#f4b840',
+  'git3': '#D26D6D'
+} } }%%
+gitGraph
+   commit tag: "v1.0"
+   commit
+   branch dev
+   checkout dev
+   commit
+   branch feature1
+   checkout feature1
+   commit
+   commit
+   checkout dev
+   branch feature2
+   checkout feature2
+   commit
+   commit
+   checkout dev
+   merge feature1
+   checkout main
+   merge dev
+   commit tag: "v2.0"
+```
+
+There are formal tools for using a Gitflow model, e.g., [nvie/gitflow](https://github.com/nvie/gitflow). However, we aren't using that tool or any others at the moment.
+
+There are components of Gitflow that we do not use. They are:
+
+- Release branches
+- Hotfix branches
```

Comment:
We do talk about using hotfix branches in the guide, and I like the concept, but I don't think we do too much that is "live" enough to require using them yet:

- https://deploy-preview-42--monumental-cassata-2f9bb7.netlify.app/codereview-dasl.html#high-level-approach:~:text=All%20Stable%20projects%20need%20branch%20protection%20on%20the%20main%20branch%20so%20that%20the%20main%20branch%20can%20only%20be%20updated%20via%20pull%20requests%20from%20dev%20or%20hotfix%2D%20branches%20with%20the%20approval%20of%20a%20project%20lead.
- https://deploy-preview-42--monumental-cassata-2f9bb7.netlify.app/maintenance.html#sec-versioning:~:text=Bug%20fixes%20and%20hotfixes%20cause%20the%20patch%20number%20to%20increment%3A%200.1.0%20%E2%86%92%200.1.1
- https://deploy-preview-42--monumental-cassata-2f9bb7.netlify.app/maintenance.html#branching:~:text=Developmental/incremental/fix/hotfix